third_party/nvfuser/csrc/ops/alias.cpp

```diff
@@ -371,4 +372,285 @@ TensorView* transpose(TensorView* x) {
   return transpose(x, 0, 1);
 }
 
+// Padding widths are assumed to be non-negative. Currently there's no
+// validation.
+TensorView* pad(TensorView* inp, const std::vector<Val*>& pad_widths) {
+  const auto& inp_dom = inp->domain()->noReductions();
+
+  const auto ndims = inp_dom.size();
+
+  TORCH_CHECK(
+      pad_widths.size() % 2 == 0 && pad_widths.size() / 2 <= ndims,
+      "Invalid number of padding widths: ",
+      pad_widths.size());
+
+  const auto num_padded_dims = pad_widths.size() / 2;
+  const auto num_non_padded_dims = ndims - num_padded_dims;
+
+  std::vector<IterDomain*> root_ids(ndims);
+  std::vector<IterDomain*> rfactor_ids(ndims);
+
+  // PadOp requires pad widths for all dimensions, even for non-padded
+  // ones.
+
+  std::vector<Val*> normalized_pad_widths;
+
+  // Fill zero for non padded dimensions
+  for (const auto i : c10::irange(num_non_padded_dims)) {
+    (void)i;
+    normalized_pad_widths.push_back(FusionGuard::getCurFusion()->zeroVal());
+    normalized_pad_widths.push_back(FusionGuard::getCurFusion()->zeroVal());
+  }
+
+  // torch.pad has padding widths of inner dimensions before outer
+  // dimensions
```

Comment:
🤢 I dislike how the Torch API has these reversed. Thanks for not maintaining this convention in PadOp.